test/script/test_ndarray.py

```diff
@@ -801,6 +801,32 @@ def generic_contiguous(x: Any) -> Any:
         self.assertEqual(z0[0][0], 1)
         self.assertEqual(z1[0][0], 1)
 
+    def test_nd_reshape(self):
+        def generic_reshape(x: Any, newshape: Any) -> Any:
+            return x.reshape(newshape)
+
+        def specific_shapre(x: matx.NDArray, newshape: tuple) -> matx.NDArray:
+            return x.reshape(newshape)
```

Comment:
Need to add a test: NDArray.reshape([...])

Reply:
added new test